### PR TITLE
refactor: enable -Wconversion warning for funcs.c and userfuncs.c

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/567

### CODE REVIEW

File: src/nvim/eval/userfunc.c [hunk 1]
```diff
@@ -278,8 +278,8 @@ int get_lambda_tv(char_u **arg, typval_T *rettv, bool evaluate)
     ga_grow(&newlines, 1);
 
     // Add "return " before the expression.
-    len = 7 + e - s + 1;
-    p = (char_u *)xmalloc(len);
+    len = (int)(7 + e - s + 1);
```
Comment (bfredl) [inline src/nvim/eval/userfunc.c:281]: is len used after this as an `int`, otherwise just `size_t len = (size_t)(7 + e - s + 1);` here

Comment (dundargoc) [reply src/nvim/eval/userfunc.c:281]: Good catch. Thanks.

File: src/nvim/eval/userfunc.c [hunk 2]
```diff
@@ -3417,7 +3416,7 @@ hashitem_T *find_hi_in_scoped_ht(const char *name, hashtab_T **pht)
   while (current_funccal != NULL) {
     hashtab_T *ht = find_var_ht(name, namelen, &varname);
     if (ht != NULL && *varname != NUL) {
-      hi = hash_find_len(ht, varname, namelen - (varname - name));
+      hi = hash_find_len(ht, varname, (size_t)((ptrdiff_t)namelen - (varname - name)));
```
Comment (bfredl) [inline src/nvim/eval/userfunc.c:3419]: `(varname - name)` is non negative so you could just `(size_t)` directly there

Comment (dundargoc) [reply src/nvim/eval/userfunc.c]: Also a good catch :)

**Review by bfredl — APPROVED** [review]
squash?

**Comment by dundargoc** [comment]
Done

**Pull request merged**